feat(search): add order venues by accessibility -option when filtering with accessibility

nikomakela opened the pull request:
Add the order by value: LIIKUNTA-568 LIIKUNTA-567. 

Connect with UnifiedSearch / venue search: LIIKUNTA-569 LIIKUNTA-570.

Continues from #518.

When the accessibility profile is selected in the search context,
the venue ordering should always be a orderByAccessibilityProfile
with the current accessibility profile.

This can be achieved by controlling the venueOrderBy-parameter of the search form.
Every time when the accessibility profile is set and it does not match the venueOrderBy,
the venueOrderBy is set and the URL is refreshed.

~~Also, when the accessibility profile is set and the ordering should be forced
to accessibility profile, the ordering selector is disabled.~~ The venueOrderBy can be rechanged to something else after accessibilityProfile selection.

<img width="1394" alt="image" src="https://github.com/City-of-Helsinki/events-helsinki-monorepo/assets/389204/e8a41239-9461-4569-8168-5dfde93520a0">

### The functions

1. Select an accessibility profile and the venue tab's order by should be forced to the value of an accessibility profile, visually shown as "Esteettömyys" and the right parameter in the URL 
<img width="550" alt="image" src="https://github.com/City-of-Helsinki/events-helsinki-monorepo/assets/389204/ae0d70c7-1220-4a07-bdcb-e198d2d3bf97">
2. Change a search tab and come back: the selection should be remembered and the value should be restored.
3. ~~The order by field should be disabled when the accessibility profile is selected. It changes back to active when the accessibility profile selection is cleared.~~ _this feature is deleted by 1 of the latest commits_

nikomakela committed:
feat(search): add order venues by accessibility -option when filtering with accessibility

LIIKUNTA-568 LIIKUNTA-567

nikomakela committed:
feat(search): add accessibility profile as an option to venue search params

LIIKUNTA-568 LIIKUNTA-569 LIIKUNTA-570

nikomakela committed:
feat(search): sync the venue search ordering with the accessibility profile

LIIKUNTA-568 LIIKUNTA-569 LIIKUNTA-570.

When the accessibility profile is selected in the search context,
the venue ordering should always be a orderByAccessibilityProfile
with the current accessibility profile.

This can be achieved by controlling the venueOrderBy-parameter of the search form.
Every time when the accessibility profile is set and it does not match the venueOrderBy,
the venueOrderBy is set and the URL is refreshed.

Also, when the accessibility profile is set and the ordering should be forced
to accessibility profile, the ordering selector is disabled.

karisal-anders commented on packages/components/src/utils/accessibilityProfile/typeguards.ts:
```diff
@@ -0,0 +1,7 @@
+import { AccessibilityProfile } from '../../types';
+
+export function isAccessibilityProfile(
+  value: string
+): value is AccessibilityProfile {
+  return Object.values<string>(AccessibilityProfile).includes(value);
+}
```
```suggestion
export function isAccessibilityProfile(
  value: unknown
): value is AccessibilityProfile {
  return Object.values(AccessibilityProfile).includes(value as AccessibilityProfile);
}
```

To make it easier to use the type guard with any input type instead of just strings.

karisal-anders commented on apps/sports-helsinki/src/domain/search/combinedSearch/adapters/VenueSearchAdapter.ts:
```diff
@@ -80,6 +82,11 @@ class VenueSearchAdapter implements CombinedSearchAdapter<VenueSearchParams> {
           ? SortOrder.Descending
           : SortOrder.Ascending,
       } as OrderByDistance;
+    } else if (
+      input.venueOrderBy &&
+      isAccessibilityProfile(input.venueOrderBy)
+    ) {
+      this.orderByAccessibilityProfile = input.venueOrderBy;
     }
```
If the suggested changes in https://github.com/City-of-Helsinki/events-helsinki-monorepo/pull/519/files#r1365036674 are taken into account then this can be simplified:
```suggestion
    } else if (isAccessibilityProfile(input.venueOrderBy)) {
      this.orderByAccessibilityProfile = input.venueOrderBy;
    }
```

nikomakela committed:
refactor: improve the isAccessibilityProfile-typeguard

LIIKUNTA-568

karisal-anders commented:
How I remember the discussion about this feature having gone is that the orderBy-dropdown shouldn't be locked if one chooses an accessibility profile "filter" in the search bar but it would remain open to be changed, serving a secondary use case of only showing the chosen accessibility profile's accessibility shortcoming counts in the search result cards but not sorting the search results by them.

Example:
- choose "I use a rollator" from the search bar's accessibility profile dropdown
- press "Search"
- this makes the orderBy-dropdown have "Accessibility" (Esteettömyys) choice in it
- the orderBy-dropdown would remain changeable
- if the user chose e.g. "Alphabetical order" the search results would be sorted by that
  - **and** the chosen accessibility profile's accessibility shortcoming counts would still be shown in the search result cards
  - **and** the chosen accessibility profile would not be reset or changed at all in the search bar or in the URL (i.e. URL parameter `accessibilityProfile`)

nikomakela committed:
feat(search): venueOrderBy can be changed after accessibilityProfile selection

LIIKUNTA-568.
It was mentioned (not written) in the refinement meeting, that it would be nice,
if the `venueOrderBy`would be set to the value of `accessibilityProfile`, when the profile
is changed and one is set, but it could be changed to something else with a reselection.
This means that the `venueOrderBy` cannot be disabled anymore.

nikomakela commented:
> How I remember the discussion about this feature having gone is that the orderBy-dropdown shouldn't be locked if one chooses an accessibility profile "filter" in the search bar but it would remain open to be changed, serving a secondary use case of only showing the chosen accessibility profile's accessibility shortcoming counts in the search result cards but not sorting the search results by them.

This should now be taken care of with https://github.com/City-of-Helsinki/events-helsinki-monorepo/pull/519/commits/bc4a1a561e177b78aae1b704fc16d33a7520f7fb.
